changelog/fragments/1678178226-Reduced-deadlocks-in-controller.yaml

```diff
@@ -0,0 +1,32 @@
+# Kind can be one of:
+# - breaking-change: a change to previously-documented behavior
+# - deprecation: functionality that is being removed in a later release
+# - bug-fix: fixes a problem in a previous version
+# - enhancement: extends functionality but does not break or fix existing behavior
+# - feature: new functionality
+# - known-issue: problems that we are aware of in a given version
+# - security: impacts on the security of a product or a user’s deployment.
+# - upgrade: important information for someone upgrading from a prior version
+# - other: does not fit into any of the other categories
+kind: bug-fix
+
+# Change summary; a 80ish characters long description of the change.
+summary: Reduced deadlocks in controller
```

Comment:
If this fixes https://github.com/elastic/elastic-agent/issues/2269 you should specifically put that it fixes a bug where the agent doesn't start monitoring new pods until the agent restarts.

This will make it really clear to affected users that the bug is supposed to be fixed. Right now a user has no idea what this fixes for them without investigating the linked issue themselves.